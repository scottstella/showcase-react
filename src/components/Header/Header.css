.header {
  width: 100%;
  background-color: #7d94b5;
  display: flex;
}

.header h1 {
  align-self: center;
}

.header-logo {
  width: 232px;
  height: 107.5px;
}
