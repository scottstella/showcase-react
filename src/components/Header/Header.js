import React from "react";

export default function Header() {
  return (
    <div class=".header">
      <h1>Showcase React</h1>
    </div>
  );
}
