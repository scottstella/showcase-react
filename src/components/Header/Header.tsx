import React from "react";
import { NavLink } from "react-router-dom";
import hearthstoneBanner from "../../assets/images/hearthstone-banner.png";
import Auth from "../Auth";
import "./Header.css";
import "../Auth.css";

const Header: React.FC = () => {
  return (
    <div className="header">
      <div className="header-left">
        <NavLink to="/">
          <img className="header-logo" src={hearthstoneBanner} alt="hearthstone-banner" />
        </NavLink>
        <h1>Deck Tracker</h1>
      </div>
      <div className="header-right">
        <Auth />
      </div>
    </div>
  );
};

export default Header;
