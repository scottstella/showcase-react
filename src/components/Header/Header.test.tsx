import React from "react";
import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Header from "./Header";

// Wrap component with Router for NavLink
const renderWithRouter = (component: React.ReactNode) => {
  return render(<BrowserRouter>{component}</BrowserRouter>);
};

describe("Header", () => {
  it("renders without crashing", () => {
    renderWithRouter(<Header />);
    expect(screen.getByText("Deck Tracker")).toBeInTheDocument();
  });

  it("has the correct CSS class", () => {
    const { container } = renderWithRouter(<Header />);
    expect(container.firstChild).toHaveClass("header");
  });

  it("renders the Hearthstone banner image", () => {
    renderWithRouter(<Header />);
    const image = screen.getByAltText("hearthstone-banner");
    expect(image).toBeInTheDocument();
    expect(image).toHaveClass("header-logo");
    expect(image.tagName.toLowerCase()).toBe("img");
  });

  it("contains a NavLink that links to home", () => {
    renderWithRouter(<Header />);
    const link = screen.getByRole("link");
    expect(link).toHaveAttribute("href", "/");
  });

  it("renders the title text", () => {
    renderWithRouter(<Header />);
    const title = screen.getByText("Deck Tracker");
    expect(title).toBeInTheDocument();
    expect(title.tagName.toLowerCase()).toBe("h1");
  });

  it("matches snapshot", () => {
    const { container } = renderWithRouter(<Header />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
