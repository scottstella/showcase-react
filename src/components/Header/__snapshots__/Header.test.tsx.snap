// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > matches snapshot 1`] = `
<div
  class="header"
>
  <a
    aria-current="page"
    class="active"
    href="/"
  >
    <img
      alt="hearthstone-banner"
      class="header-logo"
      src="/src/assets/images/hearthstone-banner.png"
    />
  </a>
  <h1>
    Deck Tracker
  </h1>
</div>
`;
